projects/abseil-cpp/string_utilities_fuzzer.cc

```diff
@@ -0,0 +1,58 @@
+// Copyright 2020 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+
+#include <string>
+
+#include <fuzzer/FuzzedDataProvider.h>
+
+#include "absl/strings/str_join.h"
+#include "absl/strings/str_split.h"
+#include "absl/strings/numbers.h"
+#include "absl/strings/str_cat.h"
+
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
+	if (size < 14)
+		return 0;
+
+	// First 4 bytes for float, next 4 for double, next 4 for int, next 1 for boolean, then atleast 1 for string 1
+	FuzzedDataProvider fuzzed_data(data, size);
+	std::string float_str = fuzzed_data.ConsumeBytesAsString(4);
```

Comment:
Just use ConsumeFloatingPoint, ConsumeIntegral etc here, then use absl::StrFormat or something similar to produce the strings, which you can pass back to SimpleAtof etc.